apps/taxes/src/modules/avatax/tax-code/avatax-tax-code-matches.router.ts

```diff
@@ -0,0 +1,41 @@
+import { createLogger } from "../../../lib/logger";
+import { AvataxTaxCodeMatchesService } from "./avatax-tax-code-matches.service";
+import { protectedClientProcedure } from "../../trpc/protected-client-procedure";
+import { router } from "../../trpc/trpc-server";
+import { avataxTaxCodeMatchSchema } from "./avatax-tax-code-match-repository";
+
+const protectedWithAvataxTaxCodeMatchesService = protectedClientProcedure.use(({ next, ctx }) =>
+  next({
+    ctx: {
+      ...ctx,
+      taxCodeMatchesService: new AvataxTaxCodeMatchesService({
+        ...ctx,
```

Comment:
Im not sure if you need to ...ctx here, its authdata that you need. And actually, like I wrote in previous comment - better add `Client` to ctx and then construct service and pass client to constructor